evm/evm-processor/src/ds-archive/schema.ts

```diff
@@ -1,29 +1,47 @@
 import {weakMemo} from '@subsquid/util-internal'
-import {array, BYTES, NAT, object, option, STRING, taggedUnion, withDefault} from '@subsquid/util-internal-validation'
-import {FieldSelection} from '../interfaces/data'
 import {
-    getBlockHeaderProps,
-    getLogProps,
-    getTraceFrameValidator,
-    getTxProps,
-    getTxReceiptProps,
-    project
-} from '../mapping/schema'
+    array,
+    BYTES,
+    NAT,
+    object,
+    option,
+    STRING,
+    taggedUnion,
+    withDefault,
+    withSentinel
+} from '@subsquid/util-internal-validation'
+import * as schema from '@subsquid/evm-data/lib/schema'
+import {FieldSelection} from '../interfaces/data'
+import {getTraceFrameValidator, project, getLogValidator} from '../mapping/schema'
 
 
 export const getBlockValidator = weakMemo((fields: FieldSelection) => {
-    let BlockHeader = object(getBlockHeaderProps(fields.block, true))
+    let BlockHeader = object({
+        ...project(fields.block, {
+            ...schema.GetBlock.props,
+            timestamp: withSentinel('BlockHeader.timestamp', 0, NAT),
+            size: withSentinel('BlockHeader.size', -1, NAT),
+            l1BlockNumber: withDefault(0, NAT),
+        }),
+        number: NAT,
+        hash: BYTES,
+        parentHash: BYTES,
+    })
 
     let Transaction = object({
-        hash: fields.transaction?.hash ? BYTES : undefined,
-        ...getTxProps(fields.transaction, true),
-        sighash: fields.transaction?.sighash ? withDefault('0x', BYTES) : undefined,
-        ...getTxReceiptProps(fields.transaction, true)
+        ...project(fields.transaction, {
```

Comment:
even though we have full schema defined we need to merge them and override fields that have different data types in archive. which is kind of ugly...